[RPA-551] Action Rework - Print to PDF

LukaszKaczkowski opened the pull request.

preludium commented on runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/Actions/browser.actions.ts:
```diff
@@ -730,11 +732,20 @@ const getBrowserActions: () => Record<string, IBpmnAction> = () => ({
+                        'ui:widget': 'TypographyWidget',
+                        'ui:options': {
+                            text: translate('Process.Details.Modeler.Actions.Browser.PrintToPdf.Session.Description'),
+                            variant: 'p',
```
there is no variant like `p`
![image](https://user-images.githubusercontent.com/44121335/203570870-c6976f4a-dbeb-4c35-aefd-d936b6f9b3b3.png)

preludium commented on runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/TypographyWidget.tsx:
```diff
@@ -0,0 +1,30 @@
+    <Box display="flex" gap="5px" alignItems="center">
+        <If condition={options.infoIcon}>
+            <InfoOutlined sx={{
+                color: (theme) => theme.palette.action.active,
```
`action.active` is meant for other purposes

use one from the grey object
![image](https://user-images.githubusercontent.com/44121335/203571332-e58013c9-9020-4b56-b472-d8a43e654e17.png)

preludium commented on runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/TypographyWidget.tsx:
```diff
@@ -0,0 +1,33 @@
+        <If condition={options.infoIcon}>
+            <InfoOutlined
+                sx={{
+                    color: blueGrey[600],
```
use theme object, we are not able to override this color in the future

preludium commented on runbotics/runbotics-orchestrator-ui/src/views/process/ProcessBuildView/Modeler/ConfigureActionPanel/widgets/TypographyWidget.tsx:
```diff
@@ -0,0 +1,33 @@
+        <If condition={options.infoIcon}>
+            <InfoOutlined
+                sx={{
+                    color: lightTheme.palette.infoIcon.default,
```
that's not how you use `theme`. Now no matter if we switch between themes you always get the same result

In case you are not able to read the docs there are 2 options

# 1
![image](https://user-images.githubusercontent.com/44121335/204275718-7612018a-8ca5-459a-99a0-29e6b20912ed.png)

# 2
![image](https://user-images.githubusercontent.com/44121335/204275777-173ebc89-e4a8-4c8f-bd1e-9f71be1f22b8.png)

preludium commented on runbotics/runbotics-orchestrator-ui/src/theme/light.ts:
```diff
@@ -1,8 +1,19 @@
+        default: string
+    }
+}
+interface ExtendedThemeOptions extends Omit<ThemeOptions, 'palette'> {
```
that's not how you do it
see this https://mui.com/material-ui/customization/theming/#custom-variables
then that `runbotics\runbotics-orchestrator-ui\src\@types\mui.d.ts`

we do it this way so the type for theme accessible inside mui components is adjusted as well and everything is type safe

preludium replied:
the best place for it is `theme.palette.info.icon` but I tried it before and it requires a walkaround so you can go with just a new field like `theme.process.action.infoIcon` yolo

LukaszKaczkowski committed:
change 'target' name to 'method',
add widget tip for session,
change 'url to print' name to 'url',
add default value for url input

LukaszKaczkowski committed:
update action names on backend

LukaszKaczkowski committed:
bump project version to 2.4.0-SNAPSHOT.2

LukaszKaczkowski committed:
refactor after cr

LukaszKaczkowski committed:
fix types

LukaszKaczkowski committed:
refactor after cr

LukaszKaczkowski committed:
add new mui color property

LukaszKaczkowski committed:
fix build

LukaszKaczkowski committed:
bump project version to 2.4.0-SNAPSHOT.3
